client/source.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package client
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// NewControllerRuntimeSource returns a reconciler for DynamicWatcher that sends events to a controller-runtime
+// source.Channel. This source.Channel can be used in the controller-runtime builder.Builder.Watches method. This
+// source.Channel will only send event.GenericEvent typed events, so any handlers specified in the
+// builder.Builder.Watches method will need to handle that.
+func NewControllerRuntimeSource(bufferSize int) (*ControllerRuntimeSourceReconciler, *source.Channel) {
+	eventChan := make(chan event.GenericEvent, bufferSize)
+	sourceChan := &source.Channel{
+		Source:         eventChan,
+		DestBufferSize: bufferSize,
```

Comment:
I wonder if we want the bufferSize to be configurable, or if it would be better for users if we just pick a good one. My first question when using this will be "what size should I use here?" and I think I'd only be able to answer that by looking at the example.

Even looking through the controller-runtime Source code, it's not immediately clear when I should change the value, or if the `eventChan` and `sourceChan` need to / should have the same buffer size...

Reply:
Okay, let's set this to the default controller-runtime of 1024.